src/Shared/Microsoft.DotNet.Scaffolding.Shared/Project/ProjectModifierHelper.cs

```diff
@@ -41,43 +42,27 @@ internal static async Task<bool> IsMinimalApp(IModelTypesLocator modelTypesLocat
         // Returns true when there is no Startup.cs or equivalent
         internal static async Task<bool> IsMinimalApp(CodeAnalysis.Project project)
         {
-            if (project != null)
+            Debugger.Launch();
+            if (project.Documents.Where(d => d.Name.EndsWith("Startup.cs")).Any())
             {
-                var startupDocument = project.Documents.Where(d => d.Name.EndsWith("Startup.cs")).FirstOrDefault();
-                if (startupDocument == null)
-                {
-                    //if changed the name in Program.cs, get the class name and check.
-                    var programDocument = project.Documents.Where(d => d.Name.EndsWith("Program.cs")).FirstOrDefault();
-                    var startupClassName = await GetStartupClassName(programDocument);
-                    if (!string.IsNullOrEmpty(startupClassName))
-                    {
-                        startupDocument = project.Documents.Where(d => d.Name.EndsWith($"{startupClassName}.cs")).FirstOrDefault();
-                    }
-                }
-
-                return startupDocument == null;
+                return false;
             }
 
-            return false;
+            // if changed the name in Program.cs, get the class name and check.
+            var programDocument = project.Documents.Where(d => d.Name.EndsWith("Program.cs")).FirstOrDefault();
+            var startupClassName = await GetStartupClassName(programDocument);
+
+            return string.IsNullOrEmpty(startupClassName); // If project has UseStartup in Program.cs, it is not a minimal app
```

Comment:
@deepchoudhery Does this make sense to you?

Reply:
this is better actually since it saves an entire loop through project files. Nice